org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -759,6 +762,14 @@ public void didChangeWorkspaceFolders(DidChangeWorkspaceFoldersParams params) {
 		return computeAsyncWithClientProgress((monitor) -> new ImplementationsHandler(preferenceManager).findImplementations(position, monitor));
 	}
 
+	@Override
+	public CompletableFuture<TypeHierarchyItem> typeHierarchy(Either<TypeHierarchyParams, ResolveTypeHierarchyItemParams> params) {
+		logInfo(">> textDocument/typeHierarchy");
```

Comment:
my understanding of the LSP proposal is we should have 2 separate requests, one to get the type hierarchy, another one to resolve type hierarchy items.  Similar to completion items. So we should have to implement 2 methods instead.